tests/test.cpp

```diff
@@ -837,6 +837,9 @@ void ValueTest() {
 
   // Test conversion functions.
   TEST_EQ(FloatCompare(TestValue("{ Y:cos(rad(180)) }"), -1), true);
+  
+  // Test negative hex constant.
+  TEST_EQ(TestValue("{ Y:-0x80 }") == -128, true);
```

Comment:
Testing `==` against floats is very fragile, please use `FloatCompare` like the other cases.

Even better would be to make `TestValue` templated so it can use both `float` and `int`, so you can do an int comparison with `==`.